contracts/core/modules/RebalancingSetIssuanceModule.sol

```diff
@@ -0,0 +1,455 @@
+     *                                     or left in the vault
+     */
+    function redeemRebalancingSetWithEther(
+        address _rebalancingSetAddress,
+        uint256 _rebalancingSetQuantity,
+        bool _keepChangeInVault
+    )
+        external
+        nonReentrant
+    {
+        // Redeem RB Set to the vault attributed to this contract
+        coreInstance.redeemModule(
+            msg.sender,
+            address(this),
+            _rebalancingSetAddress,
+            _rebalancingSetQuantity
+        );
+
+        // Calculate the Base Set Redeem quantity
+        address baseSetAddress = IRebalancingSetToken(_rebalancingSetAddress).currentSet();
+        uint256 baseSetRedeemQuantity = getBaseSetRedeemQuantity(baseSetAddress);
+
+        // Withdraw base Set to this contract
+        coreInstance.withdraw(
+            baseSetAddress,
+            baseSetRedeemQuantity
+        );
+
+        // Redeem the base Set. The components stay in the vault
+        coreInstance.redeem(
+            baseSetAddress,
+            baseSetRedeemQuantity
+        );
+
+        // Loop through the base Set components and transfer to sender.
+        withdrawComponentsToSenderWithEther(baseSetAddress);
+
+        // Transfer any change of the base Set to the end user
+        returnExcessBaseSet(baseSetAddress, _keepChangeInVault);
+
+        // Log RebalancingSetRedeem
+        emit LogRebalancingSetRedeem(
+            _rebalancingSetAddress,
+            msg.sender,
+            _rebalancingSetQuantity
+        );
+    }
+
+    /* ============ Private Functions ============ */
+
+    /**
+     * During issuance, deposit the required quantity of Base Set, wrap Ether, and deposit components
+     * (excluding Ether) to the Vault in the name of the module.
+     *
+     * @param  _baseSetAddress           Address of the base Set token
+     * @param  _baseSetQuantity          The Quantity of the base Set token to issue
+     */
+    function depositComponentsAndHandleEth(
+        address _baseSetAddress,
+        uint256 _baseSetQuantity
+    )
+        private
+    {
+        ISetToken baseSet = ISetToken(_baseSetAddress);
+
+        address[] memory baseSetComponents = baseSet.getComponents();
+        uint256[] memory baseSetUnits = baseSet.getUnits();
+        uint256 baseSetNaturalUnit = baseSet.naturalUnit();
+
+       // Loop through the base Set components and deposit components 
+        for (uint256 i = 0; i < baseSetComponents.length; i++) {
+            address currentComponent = baseSetComponents[i];
+            uint256 currentUnit = baseSetUnits[i];
+
+            // Calculate required component quantity
+            uint256 currentComponentQuantity = _baseSetQuantity.mul(currentUnit).div(baseSetNaturalUnit);
+
+            // If address is weth, deposit weth and transfer eth
+            if (currentComponent == address(weth)) {
```

Comment:
Prefer that there is a requirement that one of the required components is WETH and that ETH is wrapped and used. Maybe have this check in the very beginning. When proxy behaviors exist such as being able to use this as basically the same thing as `issueRebalancingSetWithEther` when it was not intended are when attack vectors surface.